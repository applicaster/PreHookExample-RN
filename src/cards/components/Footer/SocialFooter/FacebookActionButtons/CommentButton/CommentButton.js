import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { sendAnalyticEvent } from 'react-native-zapp-bridge';
import { COMMENT_CLICKED } from '../../../../../../constants/analyticEvents';
import ActionButton from '../../../ActionButton';
import { COMMENT_BUTTON } from '../../../../../../icons';

class CommentButton extends Component {
  constructor(props) {
    super(props);
    this.navigateToWebview = this.navigateToWebview.bind(this);
  }

  navigateToWebview() {
    const { navigation } = this.context;
    const { eventId, setActiveEventId } = this.props;

    setActiveEventId(eventId);
    navigation.navigate('SocialWebView', { headerTitle: 'Facebook' });
    sendAnalyticEvent(COMMENT_CLICKED, { eventId }).then().catch();
  }
  
  render() {
    const { commentsCount } = this.props;
    return (
      <ActionButton
        imageUri={COMMENT_BUTTON}
        label={commentsCount}
        onPress={this.navigateToWebview}
        selected={false}
      />
    );
  }
}

CommentButton.propTypes = {
  eventId: PropTypes.string.isRequired,
  commentsCount: PropTypes.number.isRequired,
  setActiveEventId: PropTypes.func.isRequired,
};

CommentButton.contextTypes = {
  navigation: PropTypes.object,
};

export default CommentButton;
