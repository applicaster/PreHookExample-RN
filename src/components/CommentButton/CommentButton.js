import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ActionButton from '../ActionButton';
import { COMMENT_BUTTON } from '../../icons';

class CommentButton extends Component {
  constructor(props) {
    super(props);
    this.navigateToWebview = this.navigateToWebview.bind(this);
  }

  navigateToWebview() {
    const { navigation } = this.context;
    const { eventId, setActiveEventId, socialNetwork } = this.props;

    setActiveEventId(eventId);
    navigation.navigate('WebView', { headerTitle: socialNetwork });
  }
  
  render() {
    const { commentsCount } = this.props;
    return (
      <ActionButton
        imageUri={COMMENT_BUTTON}
        label={commentsCount}
        onPress={this.navigateToWebview}
      />
    );
  }
}

CommentButton.propTypes = {
  eventId: PropTypes.string,
  commentsCount: PropTypes.number,
  setActiveEventId: PropTypes.func,
  socialNetwork: PropTypes.string,
};

CommentButton.contextTypes = {
  navigation: PropTypes.object,
};

export default CommentButton;
