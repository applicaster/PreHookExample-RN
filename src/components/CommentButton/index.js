import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setActiveEventId } from '../../actions';

import CommentButton from './CommentButton';

const mapDispatchToProps = dispatch => bindActionCreators({
  setActiveEventId,
}, dispatch);

export default connect(undefined, mapDispatchToProps)(CommentButton);
