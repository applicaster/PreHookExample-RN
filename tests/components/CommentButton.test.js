import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CommentButton from '../../src/components/CommentButton/CommentButton';

jest.mock('react-native-zapp-bridge', () => 'NativeZappBridge');

describe('CommentButton', () => {
  test('component renders correctly', () => {
    const tree = renderer.create(
      <CommentButton commentsCount={3} eventId={1} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
