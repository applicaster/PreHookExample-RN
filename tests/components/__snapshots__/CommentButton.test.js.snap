// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 30,
      "marginRight": 7,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "uri": "feed_rn_comment_button",
      }
    }
    style={
      Array [
        Object {
          "height": 30,
          "marginRight": 1,
          "width": 30,
        },
        Object {
          "tintColor": "#FFFFFF",
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontSize": 12,
        },
        Object {
          "color": "#FFFFFF99",
        },
      ]
    }
  >
    3
  </Text>
</View>
`;
